rbc/irtools.py

```diff
@@ -362,6 +362,27 @@ def compile_IR(ir):
     return engine
 
 
+def _fflush(builder):
+    int8_t = ir.IntType(8)
+    fflush_fnty = ir.FunctionType(int32_t, [int8_t.as_pointer()])
+    fflush_fn = builder.module.get_or_insert_function(
+        fflush_fnty, name='fflush')
+
+    builder.call(fflush_fn, [int8_t.as_pointer()(None)])
+
+
+@extending.intrinsic
+def fflush(typingctx):
+    """fflush that can be called from Numba jit-decorated functions.
+    """
+    sig = nb_types.void(nb_types.void)
+
+    def codegen(context, builder, signature, args):
+        _fflush(builder)
```

Comment:
```suggestion
@extending.intrinsic
def fflush(typingctx):
    """fflush that can be called from Numba jit-decorated functions.
    """
    sig = nb_types.void(nb_types.void)

    def codegen(context, builder, signature, args):
        int8_t = ir.IntType(8)
        fflush_fnty = ir.FunctionType(int32_t, [int8_t.as_pointer()])
        fflush_fn = builder.module.get_or_insert_function(
            fflush_fnty, name='fflush')

        builder.call(fflush_fn, [int8_t.as_pointer()(None)])

    return sig, codegen

```

Reply:
Like this? And delete `def _fflush()...`?

Reply:
exactly.

Reply:
Before, `_fflush` was compiler/platform-dependent and verbose, but know it is compiler/platform agnostic and short. Since `_fflush` is used only in codegen, it can be eliminated, imho.

Reply:
If one removes `_fflush`, we will not be able to call it inside a `codegen()`, only in the body of jitted functions. For instance:

```python
@omnisci('int32(int32)')
def foo(a):
    printf("arg: %d\n", a) # works
    flush() # works
    return a
```

```python
@extending.intrinsic
def foo(typingctx, a):
    sig = types.int32(types.int32)
    def codegen(context, builder, sig, args):
        [arg] = args
        printf("arg: %d\n", arg) # error
        fflush() # error
        # cgutils.printf(builder, "arg: %d\n", arg)
        # _fflush(builder)

        return impl_ret_untracked(context, builder, sig.return_type, arg)

    return sig, codegen
```

Reply:
Good point. Then we need two functions.

Suggestion: rename `_fflush` to `codegen_fflush` or `cg_fflush` as it is not a private function.

In the long term, perhaps `cg_fflush` should be exposed via numba as `cgutils.fflush`.

Reply:
I will add this to Numba as well

Reply:
See also the comments in https://github.com/numba/numba/issues/5679 when you are at it.